src/controllers/csi/gc/controller.go

```diff
@@ -52,55 +61,89 @@ func (gc *CSIGarbageCollector) Reconcile(ctx context.Context, request reconcile.
 	log.Info("running OneAgent garbage collection", "namespace", request.Namespace, "name", request.Name)
 	reconcileResult := reconcile.Result{RequeueAfter: 60 * time.Minute}
 
+	dynakube, err := getDynakubeFromRequest(ctx, gc.apiReader, request)
+	if err != nil {
+		return reconcileResult, err
+	}
+	if dynakube == nil {
+		return reconcileResult, nil
+	}
+
+	gcInfo, err := collectGCInfo(ctx, gc.apiReader, *dynakube)
+	if err != nil {
+		return reconcileResult, err
+	}
+	if gcInfo == nil {
+		return reconcileResult, nil
+	}
+
+	log.Info("running binary garbage collection")
+	gc.runBinaryGarbageCollection(gcInfo.pinnedVersions, gcInfo.tenantUUID, gcInfo.latestAgentVersion)
+
+	log.Info("running log garbage collection")
+	gc.runLogGarbageCollection(gcInfo.tenantUUID)
+
+	log.Info("running shared images garbage collection")
+	if err := gc.runSharedImagesGarbageCollection(); err != nil {
+		log.Info("failed to garbage collect the shared images")
+		return reconcileResult, err
+	}
+
+	return reconcileResult, nil
+}
+
+func getDynakubeFromRequest(ctx context.Context, apiReader client.Reader, request reconcile.Request) (*dynatracev1beta1.DynaKube, error) {
```

Comment:
context, apiReader and request are used together quite frequently. Would it make sense to combine them in a struct and call the functions as receiver functions instead?
Maybe something like `garbageCollectionRequest` that would result in `gcRequest.getDynakube()` or something like that